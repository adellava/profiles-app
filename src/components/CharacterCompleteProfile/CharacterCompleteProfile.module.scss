@use 'src/styles' as style;

.CharacterCompleteProfile {
  display: grid;
  grid-row-gap: 1rem;
  grid-column-gap: 1rem;
  grid-template-areas: 'img' 'description';
  grid-template-rows: auto auto;

  &_img {
    grid-area: img;
    aspect-ratio: 1/1;
    border-radius: style.$border-radius;
    object-fit: cover;
    width: 100%;
  }

  &_description {
    grid-area: description;
    text-align: left;
    display: grid;
    grid-row-gap: 1rem;
    grid-column-gap: 1rem;
    grid-template-columns: 1fr 1fr;
    grid-template-rows: auto auto auto;
    grid-template-areas: 'info info' 'location origin' 'episodes episodes';
    line-height: 1.3;
    max-width: 60ch;
  }

  &_box {
    background-color: style.$bg-color;
    border-radius: style.$border-radius;
    padding: 1.5rem;
  }

  &_info {
    grid-area: info;
  }

  &_origin {
    grid-area: origin;
  }

  &_location {
    grid-area: location;
  }

  &_episodes {
    grid-area: episodes;
  }

  &_title {
    color: style.$black;
    font-size: style.$font-size-300;
    font-weight: style.$text-featured-weight;
    margin-top: 0.3rem;
    margin-bottom: 0.8rem;
  }

  &_subtitle {
    margin-top: 0.1rem;
    margin-bottom: 0.6rem;
    color: style.$black;
    font-size: style.$font-size-400;
    font-weight: style.$text-standard-weight;
  }

  &_sectionTitle {
    color: style.$gray-600;
    text-transform: uppercase;
    font-size: style.$font-size-600;
    font-weight: style.$text-featured-weight;
    margin-bottom: 1.2rem;
    margin-top: 0.2rem;
  }
}

@include style.min-width(large) {
  .CharacterCompleteProfile {
    grid-template-areas: 'img description';
    grid-template-columns: 1fr 1.5fr;
    grid-template-rows: 1fr;
  }
}
