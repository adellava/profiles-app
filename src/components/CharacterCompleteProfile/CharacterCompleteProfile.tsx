import { CompleteCharacter, Place } from 'src/@types';
import { isAPlace } from 'src/utils';
import styles from './CharacterCompleteProfile.module.scss';

type CharacterCompleteProfileProps = {
  completeCharacter: CompleteCharacter;
};

const CharacterCompleteProfilePlace = ({
  place,
  title,
}: {
  place: Place;
  title: string;
}) => {
  return (
    <>
      <h4>{title}</h4>
      <div>{place.name}</div>
      {isAPlace(place) ? (
        <>
          <div>{place.type}</div>
          <div>{place.dimension}</div>
          <div>N residents: {place.residents.length}</div>
        </>
      ) : null}
    </>
  );
};

export const CharacterCompleteProfile = ({
  completeCharacter,
}: CharacterCompleteProfileProps) => {
  console.log(completeCharacter);

  return (
    <div className={styles.CharacterCompleteProfile}>
      <img
        className={styles.CharacterCompleteProfile_img}
        src={completeCharacter.character.image}
        alt={completeCharacter.character.name}
      ></img>
      <div className={styles.CharacterCompleteProfile_description}>
        <div
          className={`${styles.CharacterCompleteProfile_info} ${styles.CharacterCompleteProfile_box}`}
        >
          <h3>{completeCharacter.character.name}</h3>
          <div>
            <span>{completeCharacter.character.species}</span> -{' '}
            <span>{completeCharacter.character.status}</span> -{' '}
            <span>{completeCharacter.character.gender}</span>
          </div>
        </div>
        <div
          className={`${styles.CharacterCompleteProfile_origin} ${styles.CharacterCompleteProfile_box}`}
        >
          <CharacterCompleteProfilePlace
            title="Origin"
            place={completeCharacter.origin}
          />
        </div>
        <div
          className={`${styles.CharacterCompleteProfile_location} ${styles.CharacterCompleteProfile_box}`}
        >
          <CharacterCompleteProfilePlace
            title="Location"
            place={completeCharacter.location}
          />
        </div>
        <div
          className={`${styles.CharacterCompleteProfile_episodes} ${styles.CharacterCompleteProfile_box}`}
        >
          <h4>Episodes</h4>
          {completeCharacter.episodes.map((episode, i) => (
            <span key={episode.id}>
              {i !== 0 ? ', ' : null}
              {episode.name}
            </span>
          ))}
        </div>
      </div>
    </div>
  );
};
