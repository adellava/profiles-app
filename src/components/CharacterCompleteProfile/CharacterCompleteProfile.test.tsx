import { CharacterCompleteProfile } from './CharacterCompleteProfile';
import { render } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import completeProfileMock from './mock/completeProfileMock';

describe('CharacterCompleteProfile', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <CharacterCompleteProfile completeCharacter={completeProfileMock} />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should render a complete profile', () => {
    const testRenderer = TestRenderer.create(
      <CharacterCompleteProfile completeCharacter={completeProfileMock} />
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
