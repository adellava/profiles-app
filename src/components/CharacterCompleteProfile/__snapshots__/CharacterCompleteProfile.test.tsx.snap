// Vitest Snapshot v1

exports[`CharacterCompleteProfile > should render a complete profile 1`] = `
<div
  className="_CharacterCompleteProfile_bd1c2c"
>
  <img
    alt="Ants in my Eyes Johnson"
    className="_CharacterCompleteProfile_img_bd1c2c"
    src="https://rickandmortyapi.com/api/character/avatar/20.jpeg"
  />
  <div
    className="_CharacterCompleteProfile_description_bd1c2c"
  >
    <div
      className="_CharacterCompleteProfile_info_bd1c2c _CharacterCompleteProfile_box_bd1c2c"
    >
      <h3
        className="_CharacterCompleteProfile_title_bd1c2c"
      >
        Ants in my Eyes Johnson
      </h3>
      <div
        className="_CharacterCompleteProfile_subtitle_bd1c2c"
      >
        <span>
          Human
        </span>
         -
         
        <span>
          <span
            aria-label="Unknown"
            className="_StatusImage_5db57b"
            role="img"
          >
            ?
          </span>
          unknown
        </span>
         
        -
         
        <span>
          <span
            aria-label="Male"
            className="_GenderImage_d4cd8b"
            role="img"
          >
            â™‚
          </span>
          Male
        </span>
      </div>
    </div>
    <div
      className="_CharacterCompleteProfile_origin_bd1c2c _CharacterCompleteProfile_box_bd1c2c"
    >
      <h4
        className="_CharacterCompleteProfile_sectionTitle_bd1c2c"
      >
        Origin
      </h4>
      <div>
        Name: 
        unknown
      </div>
    </div>
    <div
      className="_CharacterCompleteProfile_location_bd1c2c _CharacterCompleteProfile_box_bd1c2c"
    >
      <h4
        className="_CharacterCompleteProfile_sectionTitle_bd1c2c"
      >
        Location
      </h4>
      <div>
        Name: 
        Interdimensional Cable
      </div>
      <div>
        Type: 
        TV
      </div>
      <div>
        Dimension: 
        unknown
      </div>
      <div>
        N residents: 
        62
      </div>
    </div>
    <div
      className="_CharacterCompleteProfile_episodes_bd1c2c _CharacterCompleteProfile_box_bd1c2c"
    >
      <h4
        className="_CharacterCompleteProfile_sectionTitle_bd1c2c"
      >
        Episodes
      </h4>
      <span>
        Rixty Minutes
      </span>
    </div>
  </div>
</div>
`;
